File: converters/src/main/scala/hr/element/pgscala/converters/PGNullableBigDecimalConverter.scala
package hr.element.pgscala.converters
package types

object PGNullableBigDecimalConverter extends PGTypeConverter[Option[BigDecimal]] {
  def toString(value: Option[BigDecimal]): String =
    value match {
      case Some(bD) =>
        PGBigDecimalConverter.toString(bD)
      case _ =>
        null
    }

  def fromString(value: String): Option[BigDecimal] =
    if(value != null && value.nonEmpty) {
      Some(PGBigDecimalConverter.fromString(value))
    }
    else {
      None
    }
}

File: pgjava-converters/src/main/java/hr/element/pgscala/converters/PGNullableBigDecimalConverter.java
package hr.element.pgscala.converters;

import org.joda.convert.*;

{ imports }

public enum PGNullable{ upperType }Converter implements StringConverter<BigDecimal> {
  INSTANCE;

  public static final String pgType = "decimal";

{ body }

  @ToString
  public static String { lowerType }ToString(final BigDecimal bD) {
    return null == bD ? null : bD.toString();
  }

  @FromString
  public static BigDecimal stringTo{ upperType }(final String bD) {
    return null == bD ? null : new BigDecimal(bD);
  }

// ----------------------------------------------------------------------------

  public String convertToString(final BigDecimal bD) {
    return { lowerType }ToString(bD);
  }

  public BigDecimal convertFromString(final Class<? extends BigDecimal> clazz, final String bD) {
    return stringTo{ upperName }(bD);
  }
}
